tests/test_span_session.rb

```diff
@@ -60,33 +60,95 @@ def test_session_description
     assert_equal(span.description, desc)
   end
 
-  def test_session_source_interface
+  def test_session_source_interfaces
     span = SpanSession.new(4)
     po_int = Interface.new('port-channel1')
-    ints = { 'Ethernet1/1'   => 'rx',
-             'Ethernet1/2'   => 'tx',
-             'port-channel1' => 'both',
-             'sup-eth0'      => 'rx' }
-    span.source_interface = ints
-    ints.keys.each do |int_name|
-      assert_equal(span.source_interface, int_name,
-                   "source interface #{int_name} does not match")
-    end
+    int1 = interfaces[0]
+    int2 = interfaces[1]
+    int3 = interfaces[2]
+
+    # Test default case
+    assert_equal(span.default_source_interfaces, span.source_interfaces)
+
+    # Non-default case
+    intla = { int1            => 'rx',
+              int2            => 'tx',
+              'port-channel1' => 'both',
+              'sup-eth0'      => 'rx' }
+
+    span.source_interfaces = intla
+    assert_equal(intla.to_a.sort, span.source_interfaces.sort)
+
+    # intla and intlb are identical
+    intlb = { int1            => 'rx',
+              int2            => 'tx',
+              'port-channel1' => 'both',
+              'sup-eth0'      => 'rx' }
+
+    span.source_interfaces = intlb
+    assert_equal(intlb.to_a.sort, span.source_interfaces.sort)
+
+    # intla/c same size but 1 element different
+    intlc = { int2            => 'tx',
```

Comment:
Nice find. I updated the test and it caused me to find a bug in my code :).  Bug also fixed.